sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -120,7 +122,8 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
             final SwaggerMethodParser methodParser = getMethodParser(method);
             final HttpRequest request = createHttpRequest(methodParser, args);
             Context context = methodParser.setContext(args)
-                .addData("caller-method", methodParser.getFullyQualifiedMethodName());
+                .addData("caller-method", methodParser.getFullyQualifiedMethodName())
+                .addData("eagerly-read-response", isReturnTypeDecodable(methodParser.getReturnType()));
```

Comment:
Can the user also control this behavior or is this intended for internal use only?

Reply:
The intent is for this context value being used internally but there is nothing preventing a user from creating a custom HttpPipelinePolicy that sets this context value to something else.

Reply:
Should we make this key a constant somewhere and qualify it with `azure` to minimize the chances of users using the same name in their context data (I know it's very very low)? Also, it would be good practice to have all our context data to use `azure-` prefix to differentiate from user data.